libraries/did/src/did-key-factory.ts

```diff
@@ -14,19 +14,22 @@ export class DIDKeyFactory {
    * Currently, only Ed25519 keys are supported
    * @returns The default DID Document of the DID. E.g. https://github.com/decentralized-identity/ethr-did-resolver#did-document
    */
-  public static async generate(ed25119Key: DifJsonWebKey): Promise<DIDDocument> {
-    const did = await keyToDID('key', JSON.stringify(ed25119Key));
-    const id = await keyToVerificationMethod('key', JSON.stringify(ed25119Key));
-    const verificationMethod: VerificationMethod = {
-      id,
-      type: verificationMethodTypes.Ed25519VerificationKey2018,
-      controller: did,
-      publicKeyJwk: ed25119Key
+  public static async generate(agent: Agent<{askar: AskarModule;}>, ed25119Key: DifJsonWebKey): Promise<DIDDocument> {
+    const ed25519Jwk = new Ed25519Jwk({ x: ed25119Key.x! });
+    const credoKey: Key = new Key(ed25519Jwk.publicKey, KeyType.Ed25519);
+
+    const didCreateOptions: KeyDidCreateOptions = {
+      method: 'key',
+      options: {
+        key: credoKey,
+      },
     };
 
+    const did = await agent.dids.create(didCreateOptions);
```

Comment:
@jrhender The `generate` function is coupled with the /v1/did POST API.
This particular API accepts `method and keyId ` as input parameters where keyId is optional.
1. If `keyId`(base58PublicKey) is provided, credo only allows to fetch the public key part from it's database. Therefore, we can't pass the private key.
2. If `keyId` is not passed as a parameter, the API takes care of creating the public-private key pair and then passes the public part to the `generate` function. This makes sure that the private key is available in the credo database.

I tried not changing the way we interact with the APIs, but made sure that the key for which we are creating the did actually exists in the database.

Let me know if this makes sense.

Reply:
Okay, I see that we can pass the `privateJwk` of type `Jwk` to `generate` function instead of `DifJsonWebKey`.
I think we can do that then.
I was trying to keep the function definitions as is therefore didn't think of passing the privateJwk.